### PR TITLE
Add tests and docs for more Qwen3 MoE models

### PR DESCRIPTION
**Pull request by parambole**
### Description

1. End-to-end validation scripts for the recently added `Qwen3-30B-A3B` and `Qwen3-480B-A35B` models.
2. Additionally, the documentation has been updated to serve as a guide for all additional Qwen3 MoE models.

### Key Changes

-   **New Test Scripts (`end_to_end/tpu/qwen/moe/`):**

    -   Added `qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh` to introduce a logit validation test for the 30B model.

    -   Added `qwen3-480b-a35b/1_test_qwen3_480b_a35b.sh` to introduce a logit validation test for the 480B model.

-   **Documentation Update (`end_to_end/tpu/qwen/moe/run_qwen_moe.md`):**

    -   The README has been rewritten to provide instructions for the `30B`, `235B`, and `480B` Qwen3 MoE models.

    -   The "Correctness Validation" section now has separate, easy-to-follow instructions for each model variant.

### Testing Strategy & Validation

The changes were validated by executing the new and updated test scripts.

#### **Checklist**

-   [X] I have performed a self-review of my code.
-   [X] I have necessary comments in my code, particularly in hard-to-understand areas.
-   [X] I have run end-to-end tests and provided the validation results above.
-   [X] I have made or will make corresponding changes to the documentation if needed.

### CODE REVIEW

File: end_to_end/tpu/qwen/moe/qwen3-235b-a22b/1_test_qwen3_235b_a22b.sh [hunk 1]
```diff
@@ -40,14 +40,14 @@ echo "Validating MaxText checkpoint at ${MAXTEXT_CHECKPOINT_PATH}"
 echo "Against original HF model: ${HF_MODEL_PATH}"
 
 # This command runs the core validation logic.
-JAX_PLATFORMS=cpu python3 -m MaxText.tests.forward_pass_logit_checker MaxText/configs/base.yml \
+python3 -m MaxText.tests.forward_pass_logit_checker MaxText/configs/base.yml \
```
Comment (RissyRan) [inline end_to_end/tpu/qwen/moe/qwen3-235b-a22b/1_test_qwen3_235b_a22b.sh]: So the script can be run on a TPU for larger model now?

File: end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+python3 -m MaxText.tests.forward_pass_logit_checker MaxText/configs/base.yml \
+  tokenizer_type=huggingface \
+  tokenizer_path=assets/qwen3-tokenizer \
+  megablox=False \
+  sparse_matmul=False \
+  load_parameters_path=${MAXTEXT_CHECKPOINT_PATH} \
+  model_name=qwen3-30b-a3b \
+  checkpoint_storage_concurrent_gb=512 \
+  --hf_model_path=${HF_MODEL_PATH} \
+  --max_kl_div=0.15 \
```
Comment (RissyRan) [inline end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh:53]: I noticed this value is much larger than previous criteria? 

Comparing to 
* llama3: https://github.com/AI-Hypercomputer/maxtext/blob/e31039e4e34622f5afdea48df770d4a54f057f4e/end_to_end/tpu/llama3.1/70b/3_test_llama3.1_70b.sh#L19

* deepseek2: https://github.com/AI-Hypercomputer/maxtext/blob/e31039e4e34622f5afdea48df770d4a54f057f4e/end_to_end/tpu/deepseek/v2-16b/test_deepseek.sh#L34

Is this `0.15` just following the existing value for other models or variance is a little bit high?

File: end_to_end/tpu/qwen/moe/qwen3-235b-a22b/1_test_qwen3_235b_a22b.sh (hunk 1 above)
Comment (parambole) [reply end_to_end/tpu/qwen/moe/qwen3-235b-a22b/1_test_qwen3_235b_a22b.sh]: That is a great callout. I have reverted it back so that it runs on the CPU machine. Otherwise additional configs and TPU setup would be required to run it.

File: end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh (hunk 2 above)
Comment (parambole) [reply end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh:53]: > I noticed this value is much larger than previous criteria?

Yes -- there is one particular case as discussed in https://github.com/AI-Hypercomputer/maxtext/pull/2213 where the KL divergence for a particular token is high which leads to increase in the threshold to 0.15 -- the other option that I considered back then was to use a different prompt as for all of the other prompts the diversion was very low.

File: end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# This script validates a pre-converted MaxText checkpoint against its original
+# HuggingFace counterpart to ensure numerical correctness.
+
+# ---
+# Example Usage:
+#
+# # (Required) Path to the converted MaxText checkpoint
+# export MAXTEXT_CHECKPOINT_PATH=gs://path/to/converted_ckpt/0/items/
```
Comment (hengtaoguo) [inline end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh:10]: Not mandatory, would it be better if you convert to MaxText inside this bash script, and use it for comparing logits? Similar to [test_gemma3_to_mt.sh](https://github.com/AI-Hypercomputer/maxtext/blob/main/end_to_end/tpu/gemma3/4b/test_gemma3_to_mt.sh). This could be clearer and also offer instructions on checkpoint conversion commands for moe.

**Review by hengtaoguo — APPROVED** [review]

File: end_to_end/tpu/qwen/moe/qwen3-480b-a35b/1_test_qwen3_480b_a35b.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  sparse_matmul=False \
+  load_parameters_path=${MAXTEXT_CHECKPOINT_PATH} \
+  model_name=qwen3-480b-a35b \
+  checkpoint_storage_concurrent_gb=1024 \
+  skip_jax_distributed_system=True \
+  --hf_model_path=${HF_MODEL_PATH} \
+  --max_kl_div=0.015 \
+  --run_hf_model=True
+
+echo "Validation complete."
```
Comment (RissyRan) [inline end_to_end/tpu/qwen/moe/qwen3-480b-a35b/1_test_qwen3_480b_a35b.sh]: Nit: extra line needed

**Review by RissyRan — APPROVED** [review]

File: end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh (hunk 3 above)
Comment (parambole) [reply end_to_end/tpu/qwen/moe/qwen3-30b-a3b/1_test_qwen3_30b_a3b.sh:10]: Currently I haven't done it as I still need to do additional checkpoint conversion verification for larger models like 480B -- Hence I have mentioned the existing script for performing the conversion. I will update this to use the new flow once I have verified it with 480B ( multiple runs ) and I will make further memory optimizations to it.
